Makefile

```diff
@@ -1,2 +1,151 @@
-all:
-	gcc src/main.c src/user_manual.c src/is_reach_goal.c src/win_game.c src/print_menu.c src/credit.c src/single_player.c src/game_over.c src/take_heart.c src/is_player_hit.c src/is_barrier_hit.c src/load_map.c src/update_player_pos.c src/generate_view.c src/update_barrier.c src/update_arrow.c -l sdl2 -l sdl2_ttf -l sdl2_image
+CC=gcc 
+
+LIBFLAGS= -lmingw32 -lSDL2main -lSDL2
+
+#CREATE BIN AND BUILD FOLDERS TO SAVE THE COMPILED FILES DURING RUNTIME IF THEY DO NOT EXIST
+bin_folder := $(shell mkdir -p bin) 
+build_folder := $(shell mkdir -p build) 
+results_folder := $(shell mkdir -p program_output) 
+
+#TARGET TO GENERATE THE OBJECT FILES 
+main.o: src/main.c
```

Comment:
We need to put include files as prerequisites. You can define a variable as follows: 
INCLUDES = include/map.h include/menu.h include/single_player.h include/map_textures.h
and apply changes accordingly. For example:
build\main.o: src/main.c $(INCLUDES)